OpenAPI.md

```diff
@@ -3,6 +3,43 @@
 This is a style guide for writing code in OpenAPI/Swagger using YAML. This includes all APIs created from the
 [Express API Skeleton](https://github.com/osu-mist/express-api-skeleton).
 
+## Filter Query Parameters
+
+Query parameters used for filtering data should appear in the following form:
+* `filter[<field>]=<value>`: if `field` and `value` are both a single value and the filter checks for
+  strict equality
+* `filter[<field>][<operation>]=<value>`: otherwise
+
+`field` is a semantic name for the field being used to filter the resources. If the field
+corresponds to an `attribute` of the resource, it should be named after this attribute if possible.
+`operation` refers to the operation being applied to filter the resources.
+
+Below is a list of allowed values for `operation`
+
+| `operation` | Description                                    | `field` type | `value` type |
+|-------------|------------------------------------------------|--------------|--------------|
+| `neq`       | Not equal to                                   | `* \| *[]`   | `* \| *[]`   |
+| `gt`        | Greater than                                   | `number`     | `number`     |
+| `gte`       | Greater than or equal to                       | `number`     | `number`     |
+| `lt`        | Less than                                      | `number`     | `number`     |
+| `lte`       | Less than or equal to                          | `number`     | `number`     |
+| `oneOf`     | `field` is one of the following                | `*`          | `*[]`        |
+| `some`      | `field` contains at least one of the following | `*[]`        | `*[]`        |
+| `all`       | `field` contains at least all of the following | `*[]`        | `*[]`        |
+| `fuzzy`     | `value` partially matches `field`              | `string`     | `string`     |
```

Comment:
As discussed, I think we can remove the `field` type and `value` type columns for now until we have more APIs adhere to this style and figure out more use cases.

Also, I think `gt`, `gte`, etc. can also apply to filter by the date range. For example, if we search `filter[createDate][gte]='2019-01-01'`, we should returned all resources created after `2019-01-01`. If that sounds right, can you also add it to the example?

Also, another interesting case is [the date filters of the terms API](https://github.com/osu-mist/terms-api/blob/bb6691415e8c48dc00ba85314e501a473a84f4f5/openapi.yaml#L66-L69). We are using these date filters and [the start/end dates of the terms](https://github.com/osu-mist/terms-api/blob/master/openapi.yaml#L253-L262) to filter the result. What operation will be appropriate in this case?

Reply:
From our discussion, we should probably just omit the operation for this case for now. If we can come up with a better solution in the future, we can add this to our style guide.